### ISSUE TITLE
check why workers=thread_pool is slower than threads

### ISSUE THREAD

**Issue by githubmanticore** [nck-2/test-rep#205]
According to Alexey's tests workers=thread_pool which is a default is slower than workers=threads: 
``` 
workers=thread_pool 
/json/search - 2051 rps 
/search - 2344 rps 
mysql - 2305 rps 
api - 2581 rps 
 
workers=threads 
mysql - 2924 rps 
api - 3166 rps 
``` 
 
The task is to do profiling and figure out whether we can fix it or not.

**Comment by githubmanticore** [comment]
➤ Sergey Nikolaev commented:
 

 
@glook as you're working on comparing protocols now within https://gitlab.com/manticoresearch/dev/issues/849 pls reproduce this issue too on the same test set so we're sure the issue still exists. Deeper investigation and fixing can be postponed. I'd just like to know if the issue is still there.

**Comment by githubmanticore** [comment]
➤ Sergey Nikolaev commented:
 

 
This issue is blocked by <a href ="https://gitlab.com/manticoresearch/dev/issues/946">#946</a>

**Comment by githubmanticore** [comment]
➤ Sergey Nikolaev commented:
 

 
> This issue is blocked by <a href ="https://gitlab.com/manticoresearch/dev/issues/946"><a href ="https://gitlab.com/manticoresearch/dev/issues/946">#946</a></a> 
not blocked any more as <a href ="https://gitlab.com/manticoresearch/dev/issues/946"><a href ="https://gitlab.com/manticoresearch/dev/issues/946">#946</a></a> is done
